pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2720,6 +2722,25 @@ def test_merge_different_index_names():
     tm.assert_frame_equal(result, expected)
 
 
+@td.skip_if_no("dask")
+def test_merge_on_arraylike_deprecation():
+    # deprecate allowing non-standard array-likes for "on"
+
+    left = DataFrame({"A": range(3), "B": range(1, 4)})
+    right = DataFrame({"C": range(2, 5)})
+
+    import dask.array
+
+    arr = dask.array.array([0, 1, 2])  # matches left["A"]
+
+    msg = "will only allow array-like objects that are one of the following types"
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        res = merge(left, right, left_on="A", right_on=arr)
```

Comment:
Could you also add tests that the FutureWarning shows up when the dask array is passed to `left_on` and `on`?

Reply:
huh this now raises for me locally; i wonder if it isnt getting run on the CI?  ive gotta run, will dig into this later